### PR TITLE
Basic doc on msbuild and the gac

### PR DESCRIPTION
**Pull request by BenVillalobos**
### Context
I'd like to have an easy link to send folks that have GAC issues with msbuild. aka.ms/msbuild/ungac should do the trick. There's no good place to point that link to, so here's a basic doc that should get folks unblocked if they really are seeing a GAC issue with MSBuild.

### Changes Made
Added a doc.

### CODE REVIEW

File: documentation/wiki/UnGAC.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# MSBuild, the Global Assembly Cache (GAC), and You
+
+## What is the GAC?
+
+See the [public documentation](https://docs.microsoft.com/dotnet/framework/app-domains/gac). The GAC is a folder where different installations of VS on the same machine look for assemblies that are commonly used. If an assembly is in the GAC, it will be prioritized over any other assembly.
+
+The only MSBuild assemblies you may see in the GAC are version 4.0. There is no reason any modern MSBuild assembly should be in the GAC today.
```
Comment (Forgind) [inline documentation/wiki/UnGAC.md:7]: Isn't it version 4.8?

Comment (Forgind) [reply documentation/wiki/UnGAC.md:7]: And maybe define "modern" (15.0+)

File: documentation/wiki/UnGAC.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## What MSBuild Assemblies are installed on my Machine?
+Run the [EnumerateMSBuild powershell script](https://github.com/Microsoft/msbuild/blob/master/scripts/EnumerateMSBuild.ps1) from our repo. It will output a `msbuild_versions.txt` file that lists MSBuild assemblies in their common folders along with their versions.
+
+## How to Remove MSBuild Assemblies from the GAC
+1. If on Visual Studio 16.8 or higher, repair your installation.
```
Comment (Forgind) [inline documentation/wiki/UnGAC.md:13]: This sounds like `and`, whereas I'm assuming you mean `or`.

**Review by Forgind — COMMENTED** [review]
And maybe also add a comment about other GAC-related issues we've had, since other assemblies in the GAC can mess with MSBuild. I'm particularly thinking of the System.Memory problem, but a variety of other assemblies we rely on could theoretically have the same problem.

It might also be worth emphasizing that putting something like MSBuild in the GAC seems like a solution because it makes problems go away, but it causes more problems down the road when you've forgotten you did that, and the next version comes out, and you get hard-to-understand version mismatch errors.

File: documentation/wiki/UnGAC.md (hunk 1 above)
Comment (BenVillalobos) [reply documentation/wiki/UnGAC.md:7]: You're right, folders are marked as 4.0 but the assemblies themselves are 4.8.4

**Comment by BenVillalobos** [comment]
@Forgind Do you by chance have a link to a system.memory issue we've fixed by removing it from the GAC?

**Comment by Forgind** [comment]
This took too long to find:
https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1249222

**Comment by BenVillalobos** [comment]
@Forgind Much appreciated. I posted it in [our tracking issue](https://github.com/dotnet/msbuild/issues/5955). We can keep posting related assemblies there so we don't need to update this doc much.

**Review by Forgind — APPROVED** [review]

**Pull request merged**